agent/nmap_agent.py

```diff
@@ -374,16 +374,26 @@ def _emit_fingerprints(
                     raise ValueError(f"Incorrect ip version {version}")
 
                 if host.get("os", {}).get("osmatch") is not None:
-                    os_match_highest = host.get("os").get("osmatch", {})[0]
+                    os_match = host.get("os").get("osmatch", [])
```

Comment:
```suggestion
                    os_match = host.get("os").get("osmatch")
```

Reply:
I understand, but the check you have in the previous, makes it so that it's impossible for `osmatch` to not be found in this line

Reply:
done